syntax = "proto3";

package payment;

service PaymentService {
  rpc ProcessPayment(PaymentRequest) returns (PaymentResponse);
  rpc GetPayment(PaymentDetailsRequest) returns (PaymentResponse);
}

message PaymentRequest {
  string order_id = 1;
  string user_id = 2;
  double amount = 3;
  string payment_method = 4;
  string card_number = 5;
  string expiry_date = 6;
  string cvv = 7;
}

message PaymentDetailsRequest {
  string payment_id = 1;
}

message PaymentResponse {
  string payment_id = 1;
  string order_id = 2;
  string user_id = 3;
  double amount = 4;
  string status = 5;
  string payment_method = 6;
  string transaction_id = 7;
  string message = 8;
}
